Sources/Bot/Services/MergeService.swift

```diff
@@ -751,18 +695,56 @@ extension MergeService.State {
     }
 }
 
-// MARK: - Helpers
+// MARK: - Healthcheck
 
-private extension PullRequest {
+extension MergeService {
+    public final class Healthcheck {
 
-    func isLabelled(with label: PullRequest.Label) -> Bool {
-        return labels.contains(label)
-    }
-    func isLabelled(withOneOf possibleLabels: [PullRequest.Label]) -> Bool {
-        return labels.contains(where: possibleLabels.contains)
+        public enum Reason: Error, Equatable {
+            case potentialDeadlock
+        }
+
+        public enum Status: Equatable {
+            case ok
+            case unhealthy(Reason)
+        }
+
+        public let status: Property<Status>
+
+        internal init(
+            state: Signal<State, NoError>,
+            statusChecksTimeout: TimeInterval,
+            scheduler: DateScheduler
+        ) {
+            status = Property(
+                initial: .ok,
+                then: state.combinePrevious()
+                    .skipRepeats { lhs, rhs in
```

Comment:
`skipRepeats()` should be enough no?

Reply:
I remember asking that to @dmcrodrigues  too back in the day when I saw this code (which is not new, was just moved in this PR from one place to another), and he gave me a good justification for it at that point, but I admit I can't remember the rationale 😅 (which means it would be worth a comment once we remember the reason again)

Reply:
Still using Swift 4.2 and because we applied `combinePrevious` we are dealing with tuples that do not conform to `Equatable`.

Reply:
Comment added in 128eb64cceb642a8d9492fa5a5b082aa6d23c897 about this.